### DISCUSSION TITLE
collect duplicates under 1 array

### DISCUSSION THREAD

**Discussion by achursin1986** [nlohmann/json#3963] [Q&A]
Hi,
seems very similar thing was asked here: Handling Duplicate keys in json #3392, I am just looking for some examples how to implement it via callback, in the old thread there were no examples provided unfortunately .

I have a file with example simple.json

./json-test
key "reachability-tlv" was already seen in this object!
key "reachability-tlv" was already seen in this object! <--- so we see this is not unique, just need to append contents to first found object in the *same hierarchy*, which is array 

 "reachability-tlv" : [
                    { }  { }   { }   + another reachability-tlv { } { } { }  ...  ] 
                    

Just to make clarity : 

leaf [isis-tlv] can contain several [reachability-tlv] and I am trying to make them as one, no need to collapse reachability-tlv from another part of json , only local hierarchy   

Thanks in advance !
                   
p.s. full file is here https://file.io/eaHJnrmUMeuE

**Comment by nlohmann** [comment]
This is the same underlying limitation as discussion #3392: json_sax_dom_parser's object handling calls key() then start_object()/individual value callbacks with no built-in hook that says 'this key already exists in the current object, merge instead of overwrite.' The practical approach is to implement your own SAX consumer (deriving from nlohmann::json_sax<json>, see docs/mkdocs/docs/features/parsing/sax_interface.md) that tracks the currently-open object's keys itself: in your key(string_t& val) override, if val matches the key you care about (e.g. 'reachability-tlv') and it was already seen at this nesting depth, redirect subsequent value/object/array events to append into an existing array instead of starting a new key — this requires manually maintaining a small stack of 'seen keys per open object,' since the default DOM parser has no duplicate-key-merge mode.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
